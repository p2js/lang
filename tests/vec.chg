
type Vec<T> constructible(mkVec) = {
    ptr: T*,
    len: i32,
    cap: i32
};

function mkVec<T>(): Vec<T> {
    return .{ ptr: 0, len: 0, cap: 0 };
}

function main(): i32 {
    return 0;
}
