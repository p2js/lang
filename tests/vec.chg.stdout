expr @ 9:20 9:21 => i32
expr @ 9:28 9:29 => i32
expr @ 9:36 9:37 => i32
expr @ 9:12 9:39 => {ptr: i32,len: i32,cap: i32}
expr @ 9:5 9:39 => never
expr @ 8:29 11:0 => void
expr @ 13:12 13:13 => i32
expr @ 13:5 13:13 => never
expr @ 12:22 15:1 => void

define i32 @main() {
prologue:
%l.0 = alloca i32
br label %bb.0

bb.0:
store i32 0, i32* %l.0
%ret.0 = load i32, i32* %l.0
ret i32 %ret.0

}


